### PR TITLE
TILA-1355 | CSV export of reservation units

### PR DESCRIPTION
**Pull request by Niilis**
Export data of reservation units onto CSV file for easy inspection.

Refs TILA-1355

### CODE REVIEW

File: reservation_units/tests/test_reservation_unit_export.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+from django.test.testcases import TestCase
+from django.utils import timezone
+
+from ..utils.export_data import ReservationUnitExporter
+from .factories import ReservationUnitFactory
+
+
+class ReservationUnitDataExporterTestCase(TestCase):
+    export_dir = Path(settings.BASE_DIR) / "exports" / "reservation_unit_exports"
+    file_name = f"reservation_units__{timezone.now().strftime('%d-%m-%Y')}.csv"
```
Comment (nicobav) [inline reservation_units/tests/test_reservation_unit_export.py:17]: now my eyes are starting to see these double underscores :eyes:

File: reservation_units/tests/test_reservation_unit_export.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.reservation_unit = ReservationUnitFactory()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+
+        if self.export_dir.is_dir():
+            shutil.rmtree(self.export_dir)
```
Comment (nicobav) [inline reservation_units/tests/test_reservation_unit_export.py:27]: Thing to consider; If we would be using file mock here we wouldn't actually need to write anything and cleanup things.

File: reservation_units/utils/export_data.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    @classmethod
+    def export_reservation_unit_data(cls):
+        now = timezone.now()
+        root = Path(settings.BASE_DIR)
+        path = root / "exports" / "reservation_unit_exports"
+        path.mkdir(parents=True, exist_ok=True)
+        reservation_units: List[ReservationUnit] = list(ReservationUnit.objects.all())
+
+        if reservation_units:
+            file_name = f"reservation_units__{now.strftime('%d-%m-%Y')}.csv"
```
Comment (nicobav) [inline reservation_units/utils/export_data.py:21]: double underscore in the filename - intended?

**Review by nicobav — APPROVED** [review]
This is good as is, nice :cookie:

Couple comments which does not require actions.

File: reservation_units/tests/test_reservation_unit_export.py (hunk 1 above)
Comment (Niilis) [reply reservation_units/tests/test_reservation_unit_export.py:17]: Yeah, this is one of those "up to preference" things :smile: .

I used double underscores in the filename to specifically differentiate between name and date. There are many naming conventions, that could be used.

```
reservation_units__10-3-2022.csv
reservation-units_10-3-2022.csv
"reservation-units 10-3-2022.csv"
ReservationUnits_10-3-2022.csv
etc...
```

What would you say, is the one I chose good or should these be changed?

Comment (nicobav) [reply reservation_units/tests/test_reservation_unit_export.py:17]: This is a taste thing and nothing to be changed - I would just said the file to be `reservation_units_10-3-2022.csv` 
No need to change, the one you chose is good :heavy_check_mark:  =>  lets merge :shipit:

**Pull request merged**